### PR TITLE
Parameterising http(s) in soef connection

### PR DESCRIPTION
**Pull request by 5A11**
This PR parameterises http/https in the soef connection.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/soef/connection.py [hunk 1]
```diff
@@ -1286,21 +1292,29 @@ def __init__(self, **kwargs: Any) -> None:
                 self.DEFAULT_CONNECTION_CHECK_MAX_RETRIES,
             ),
         )
+        is_https = cast(bool, self.configuration.config.get("is_https"))
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:1295]:
```suggestion
        is_https = cast(bool, self.configuration.config.get("is_https"), True)
```
Makes sense to just set a default for backwards compatibility in tests etc.

File: packages/fetchai/connections/soef/connection.py [hunk 2]
```diff
@@ -1286,21 +1292,29 @@ def __init__(self, **kwargs: Any) -> None:
...
+        not_none_params = {
+            "api_key": api_key,
+            "soef_addr": soef_addr,
+            "soef_port": soef_port,
+        }
+        for param_name, param_value in not_none_params.items():  # pragma: nocover
+            if param_value is None:
+                raise ValueError(f"{param_name} must be set!")
         self.api_key = api_key
+        self.is_https = is_https if is_https is not None else True
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:1311]: None Pythonic, also see above we can set default

Comment (5A11) [reply packages/fetchai/connections/soef/connection.py:1311]:
```suggestion
        self.is_https = is_https
```

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
lgtm

**Pull request merged**